scapy/contrib/automotive/someip.py

```diff
@@ -221,3 +235,310 @@ def _bind_someip_layers():
+    fields_desc = [
+        _SDOption_Header,
+        ShortField("priority", 0),
+        ShortField("weight", 0)
+    ]
+
+
+class SDOption_IP4_EndPoint(_SDOption_IP4):
+    name = "IP4 EndPoint Option"
+    _defaults = {'type': _SDOption.IP4_ENDPOINT_TYPE,
+                 'len': _SDOption.IP4_ENDPOINT_LEN}
+
+
+class SDOption_IP4_Multicast(_SDOption_IP4):
+    name = "IP4 Multicast Option"
+    _defaults = {'type': _SDOption.IP4_MCAST_TYPE,
+                 'len': _SDOption.IP4_MCAST_LEN}
+
+
+class SDOption_IP4_SD_EndPoint(_SDOption_IP4):
+    name = "IP4 SDEndPoint Option"
+    _defaults = {'type': _SDOption.IP4_SDENDPOINT_TYPE,
+                 'len': _SDOption.IP4_SDENDPOINT_LEN}
+
+
+class SDOption_IP6_EndPoint(_SDOption_IP6):
+    name = "IP6 EndPoint Option"
+    _defaults = {'type': _SDOption.IP6_ENDPOINT_TYPE,
+                 'len': _SDOption.IP6_ENDPOINT_LEN}
+
+
+class SDOption_IP6_Multicast(_SDOption_IP6):
+    name = "IP6 Multicast Option"
+    _defaults = {'type': _SDOption.IP6_MCAST_TYPE,
+                 'len': _SDOption.IP6_MCAST_LEN}
+
+
+class SDOption_IP6_SD_EndPoint(_SDOption_IP6):
+    name = "IP6 SDEndPoint Option"
+    _defaults = {'type': _SDOption.IP6_SDENDPOINT_TYPE,
+                 'len': _SDOption.IP6_SDENDPOINT_LEN}
+
+
+##
+# SD PACKAGE DEFINITION
+##
+class SD(_SDPacketBase):
+    """
+    SD Packet
+
+    NOTE :   when adding 'entries' or 'options', do not use list.append()
+        method but create a new list
+    e.g. :  p = SD()
+            p.option_array = [SDOption_Config(),SDOption_IP6_EndPoint()]
+    """
+    SOMEIP_MSGID_SRVID = 0xffff
+    SOMEIP_MSGID_SUBID = 0x1
+    SOMEIP_MSGID_EVENTID = 0x100
+    SOMEIP_PROTO_VER = 0x01
+    SOMEIP_IFACE_VER = 0x01
+    SOMEIP_MSG_TYPE = SOMEIP.TYPE_NOTIFICATION
+
+    name = "SD"
+    _sdFlag = collections.namedtuple('Flag', 'mask offset')
+    FLAGSDEF = {
+        "REBOOT": _sdFlag(mask=0x80, offset=7),
+        "UNICAST": _sdFlag(mask=0x40, offset=6)
+    }
+
+    name = "SD"
+    fields_desc = [
+        ByteField("flags", 0),
+        X3BytesField("res", 0),
+        FieldLenField("len_entry_array", None,
+                      length_of="entry_array", fmt="!I"),
+        PacketListField("entry_array", None, cls=_SDEntry,
+                        length_from=lambda pkt: pkt.len_entry_array),
+        FieldLenField("len_option_array", None,
+                      length_of="option_array", fmt="!I"),
+        PacketListField("option_array", None, cls=_SDOption,
+                        length_from=lambda pkt: pkt.len_option_array)
+    ]
+
+    def get_flag(self, name):
+        name = name.upper()
+        if name in self.FLAGSDEF:
+            return ((self.flags & self.FLAGSDEF[name].mask) >>
+                    self.FLAGSDEF[name].offset)
+        else:
+            return None
+
+    def set_flag(self, name, value):
+        name = name.upper()
+        if name in self.FLAGSDEF:
+            self.flags = (self.flags &
```

Comment:
Can you split this on different lines?
For example
```
self.flags &= ctypes.c_ubyte(~self.FLAGSDEF[name].mask).value)
self.flags |= (value & 0x01) << self.FLAGSDEF[name].offset
```

Reply:
I didn't write this part of code myself and I don't really understand what it does. If you are sure these two lines will work I will replace the actual line.